release.sh

```diff
@@ -17,17 +19,23 @@ docker tag "${IMAGE_NAME}" "${image_tag}"
 
 echo "Tagged docker image $image_tag"
 
+
 # If the branch is master, publish it.
 if [ "$ref" = "master" ]; then
-  # Login to Docker
-  echo "${DOCKER_REGISTRY_PASSWORD}" | docker login -u "${DOCKER_REGISTRY_USERNAME}" --password-stdin
-
-  # Push the newly tagged image to registry and GitHub
-  docker push "${image_tag}"
-  docker push "${IMAGE_NAME}:latest"
-  hub release create "${new_version}" -m "${new_version}" || true
-
-  echo "Published $image_tag"
+  for Item in ${sensible_files[*]} ;
+  do
+    compare_items=$(git diff --name-only HEAD~1..HEAD | grep $Item | grep -v example)
+    if [ "$Item" = "$compare_items" ] ; then
+      # Login to Docker
+      echo "${DOCKER_REGISTRY_PASSWORD}" | docker login -u "${DOCKER_REGISTRY_USERNAME}" --password-stdin
+      # Push the newly tagged image to registry and GitHub
+      docker push "${image_tag}"
+      docker push "${IMAGE_NAME}:latest"
+      hub release create "${new_version}" -m "${new_version}" || true
+      echo "Publish skipped"
+      continue
+    fi
+  done
```

Comment:
@enflo Looks like you're trying to release the image inside a loop. I think the comparison and checking should be done before building and releasing the image even. You can do the check once - before building the docker image even and only if build the image if the condition holds true.

Let me know if it makes sense; feel free to ask if you have any queries :).

Reply:
Makes sense. I'd agree with your last suggestion - extract a function to check and you can do this check at the top around line 12. So, that if the validation fails (that is image doesn't need to be released), you can just skip the later lines.